internal/testutil/storage_cleaner.go

```diff
@@ -28,15 +28,43 @@ import (
 	"google.golang.org/api/iterator"
 )
 
-// CreateTestBucket creates a new bucket with the given prefix
-func CreateTestBucket(ctx context.Context, t *testing.T, client *storage.Client, projectID, prefix string) (string, error) {
+// TestBucket creates a new bucket with the given prefix and registers a cleanup
+// function to delete the bucket and any objects it contains when the test finishes.
+// TestBucket returns the bucket name. It fails the test if bucket creation fails.
+func TestBucket(ctx context.Context, t *testing.T, projectID, prefix string) string {
+	t.Helper()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	t.Cleanup(func() { client.Close() })
+	return CreateTestBucket(ctx, t, client, projectID, prefix)
+}
+
+// CreateTestBucket creates a new bucket with the given prefix and registers a
+// cleanup function to delete the bucket and any objects it contains.
+// It is equivalent to TestBucket but allows Storage Client re-use.
+func CreateTestBucket(ctx context.Context, t *testing.T, client *storage.Client, projectID, prefix string) string {
 	t.Helper()
 	bucketName := UniqueBucketName(prefix)
-	return bucketName, cleanBucketWithClient(ctx, t, client, projectID, bucketName)
+
+	b := client.Bucket(bucketName)
+	if err := b.Create(ctx, projectID, nil); err != nil {
+		t.Fatalf("Bucket.Create(%q): %v", bucketName, err)
+	}
+
+	t.Cleanup(func() {
+		if err := DeleteBucketIfExists(ctx, client, bucketName); err != nil {
+			log.Printf("Bucket.Delete(%q): %v", bucketName, err)
+		}
+	})
+	return bucketName
 }
 
 // CleanBucket creates a new bucket. If the bucket already exists, it will be
 // deleted and recreated.
+// Deprecated: use TestBucket or CreateTestBucket instead.
```

Comment:
Just FYI, there are a bunch of uses of this outside of GCS samples -- those will have to be replaced as well.

Reply:
Yes, I have a follow up ready to replace those.